pkg/infrastructure/clusterapi/clusterapi.go

```diff
@@ -198,17 +209,27 @@ func (i *InfraProvider) Provision(ctx context.Context, dir string, parents asset
 			Cap:      timeout,
 		}, func(ctx context.Context) (bool, error) {
 			c := &clusterv1.Cluster{}
-			if err := cl.Get(ctx, client.ObjectKey{
-				Name:      clusterID.InfraID,
-				Namespace: capiutils.Namespace,
-			}, c); err != nil {
-				if apierrors.IsNotFound(err) {
-					return false, nil
+			var clusters []*clusterv1.Cluster
+			ready := true
+			for _, curClusterID := range clusterIDs {
+				if err := cl.Get(ctx, client.ObjectKey{
+					Name:      curClusterID,
+					Namespace: capiutils.Namespace,
+				}, c); err != nil {
+					if apierrors.IsNotFound(err) {
+						return false, nil
+					}
+					return false, err
 				}
-				return false, err
+				clusters = append(clusters, c)
 			}
-			cluster = c
-			return cluster.Status.InfrastructureReady, nil
+
+			for _, curCluster := range clusters {
+				ready = ready && curCluster.Status.InfrastructureReady
```

Comment:
Not sure if it's worth it, but we could short-circuit here:
```suggestion
				if !curCluster.Status.InfrastructureReady {
				    return false, nil
				}
```

Reply:
I thought about it but wasn't sure the implications.  I can add that for sure since you think its good to have.

Reply:
makes sense.  thanks!